src/object/object_representation.c

```diff
@@ -309,14 +317,41 @@ or_class_name (RECDES * record)
   len = (int) *((unsigned char *) start);
   if (len != 0xFF)
     {
-      name = start + 1;
+      *string = start + 1;
```

Comment:
Looks like the following code is missing?
`*alloced_string = 0;`

Reply:
Yes. I forgot that asserts can only be checked in debug mode.